spec/controllers/v1.0/workflows_controller_spec.rb

```diff
@@ -41,23 +65,46 @@
   end
 
   describe 'GET /workflows' do
-    before { get "#{api_version}/workflows", :params => { :limit => 5, :offset => 0 }, :headers => request_header }
-
-    context 'when no relate wiht template'
-    it 'returns status code 200' do
-      expect(response).to have_http_status(200)
+    before do
+      allow(RBAC::Access).to receive(:new).with('workflows', 'read').and_return(access_obj)
+      allow(access_obj).to receive(:process).and_return(access_obj)
     end
 
-    it 'returns all template workflows' do
-      expect(json['links']).not_to be_empty
-      expect(json['links']['first']).to match(/limit=5&offset=0/)
-      expect(json['links']['last']).to match(/limit=5&offset=15/)
-      expect(json['data'].size).to eq(5)
+    context 'admin role return workflows' do
+      let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => true, :approver? => false, :owner? => false) }
+
+      before do
+        allow(access_obj).to receive(:not_owned?).and_return(false)
+        allow(access_obj).to receive(:not_approvable?).and_return(false)
+        allow(access_obj).to receive(:approver_id_list).and_return([])
+        allow(access_obj).to receive(:owner_id_list).and_return([])
+
+        get "#{api_version}/workflows", :params => { :limit => 5, :offset => 0 }, :headers => request_header
+      end
+
+      it 'returns status code 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'returns all workflows' do
+        expect(json['links']).not_to be_empty
+        expect(json['links']['first']).to match(/limit=5&offset=0/)
+        expect(json['links']['last']).to match(/limit=5&offset=15/)
+        expect(json['data'].size).to eq(5)
+      end
     end
   end
 
   describe "GET /workflows with filter" do
+    let(:access_obj) { instance_double(RBAC::Access, :accessible? => true, :admin? => true, :approver? => false, :owner? => false) }
     before do
+      allow(RBAC::Access).to receive(:new).with('workflows', 'read').and_return(access_obj)
+      allow(access_obj).to receive(:process).and_return(access_obj)
+      allow(access_obj).to receive(:not_owned?).and_return(false)
+      allow(access_obj).to receive(:not_approvable?).and_return(false)
+      allow(access_obj).to receive(:approver_id_list).and_return([])
+      allow(access_obj).to receive(:owner_id_list).and_return([])
+
```

Comment:
@hsong-rh Are the limit and offset required parameters in the line below, couldn't we just use the defaults